server/controllers/classifier.js

```diff
@@ -168,55 +192,57 @@ class Classifier {
-        let performedLookahead = false;
-        let labeledTokenInfo = Classifier.getMostLikelyMatch(tokenHolder.token, labelInfos);
-        if (labeledTokenInfo && labeledTokenInfo.distance < tokenHolder.distance) {
+        let labeledTokenInfo = this.getMostLikelyMatch(labelInfos);
+        if (labeledTokenInfo && labeledTokenInfo.distance <= tokenHolder.distance && (tokenHolder.label == null || labeledTokenInfo.label.length >= tokenHolder.label.length)) {
             // this layer matches for this token
             this.state.tokenHolders[this.state.currentToken] = labeledTokenInfo;
-        } else if (lookahead && this.state.currentToken <= this.state.tokenHolders.length - 2) {
-            let nextTokenHolder = this.state.tokenHolders[this.state.currentToken + 1];
-            labeledTokenInfo = Classifier.getMostLikelyMatch(tokenHolder.token + " " + nextTokenHolder.token, labelInfos);
-            if (labeledTokenInfo && labeledTokenInfo.distance < tokenHolder.distance) {
-                performedLookahead = true;
-                //replace the currentToken
-                this.state.tokenHolders[this.state.currentToken] = labeledTokenInfo;
-            }
+
         }
-        this.nextAction(performedLookahead);
+        this.nextAction();
     }
 
     /**
      * gets the most likely match for a token out of an array of values
-     * @param token token to match
      * @param labelInfos possible values which it could match
      * @returns {*}
      */
-    static getMostLikelyMatch(token, labelInfos) {
+    getMostLikelyMatch(labelInfos) {
         let labeledTokenInfos = [];
         labelInfos.forEach(lInfo => {
-            let distance;
             if (lInfo.synonyms && lInfo.synonyms.length !== 0) {
                 //get best fitting synonym
                 let labelVariation = [];
-                labelVariation [0] = lInfo.label;
+                labelVariation [0] = [lInfo.label, lInfo.token];
                 labelVariation = labelVariation.concat(lInfo.synonyms);
-                let bestSynonym = Classifier.getMostLikelyMatch(token, labelVariation.map(lV => Classifier.createLabelInfo(lV, lInfo.labelType, lInfo.datatype, lInfo.column)));
-                distance = bestSynonym.distance;
+                let bestSynonym = this.getMostLikelyMatch(labelVariation.map(lV => Classifier.createLabelInfo(lV, lInfo.labelType, lInfo.datatype, lInfo.column)));
+                labeledTokenInfos.push(Classifier.createLabeledTokenInfo(this.state.tokenHolders[this.state.currentToken].token, bestSynonym.distance, lInfo));
             } else {
                 let value = (lInfo.datatype === Classifier.col_types.date) ? lInfo.label.toString() : lInfo.label;
-                distance = Classifier.getLevenshteinDistance(token, value);
+                if (value instanceof Array && value.length > 0) {
+                    value = value[0]
+                }
+                let wordCount = value.split(' ').length;
```

Comment:
word count is irritating, because you are counting the spaces in between the words, word count would be 
``` wordCount = value.split(' ').length + 1 ```
maybe wordSeparatorCount?

Reply:
no, im actually counting words
```
let hi_tobi = 'hi tobi'.split(' ') // -> ['hi', 'tobi']
let wordVound = hi_tobi.length // -> 2
```

Reply:
yeah you are right 🥇